arangod/RocksDBEngine/RocksDBBuilderIndex.cpp

```diff
@@ -167,24 +442,80 @@ Result RocksDBBuilderIndex::remove(transaction::Methods& trx,
   return Result();  // do nothing
 }
 
+static Result processPartitions(
+    bool isForeground, std::deque<std::pair<uint64_t, uint64_t>> partitions,
+    trx::BuilderTrx& trx, rocksdb::Snapshot const* snap,
+    RocksDBCollection* rcoll, rocksdb::DB* rootDB, RocksDBIndex& ridx,
+    std::atomic<uint64_t>& docsProcessed, size_t numThreads,
+    uint64_t threadBatchSize, rocksdb::Options const& dbOptions,
+    std::string const& idxPath) {
+  auto sharedWorkEnv =
+      std::make_shared<SharedWorkEnv>(std::move(partitions), rcoll->objectId());
+  std::vector<std::unique_ptr<IndexCreatorThread>> idxCreatorThreads;
+  for (size_t i = 0; i < numThreads; ++i) {
+    auto newThread = std::make_unique<IndexCreatorThread>(
+        false, isForeground, threadBatchSize, docsProcessed, sharedWorkEnv,
+        rcoll, rootDB, ridx, snap, dbOptions, idxPath);
+    idxCreatorThreads.emplace_back(std::move(newThread));
+  }
+
+  try {
+    for (auto& idxCreatorThread : idxCreatorThreads) {
+      idxCreatorThread->start();
```

Comment:
`Thread::start()` returns a boolean result, which is not handled here.

Reply:
done